services/hockeyapp/hockeyapp.service.js

```diff
@@ -0,0 +1,83 @@
+'use strict'
+
+const Joi = require('joi')
+const { renderVersionBadge } = require('../../lib/version')
+const BaseJsonService = require('../base-json')
+const { NotFound } = require('../errors')
+
+const hockeyappSchema = Joi.object({
+  app_versions: Joi.array().items(
+    Joi.object({
+      version: Joi.string().required(),
+      shortversion: Joi.string().required(),
+    })
+  ),
+}).required()
+
+module.exports = class Hockeyapp extends BaseJsonService {
+  async fetch({ apptoken, appid }) {
+    const url = `https://rink.hockeyapp.net/api/2/apps/${appid}/app_versions`
+    const options = { headers: { 'X-HockeyAppToken': apptoken } }
+    return this._requestJson({
+      url,
+      schema: hockeyappSchema,
+      options: options,
+    })
+  }
+
+  static render({ version }) {
+    return renderVersionBadge({ version })
+  }
+
+  async handle({ apptoken, appid }) {
+    const json = await this.fetch({ apptoken, appid })
+
+    if (json.app_versions === undefined || json.app_versions.length === 0) {
+      throw new NotFound()
+    }
+
+    const latestVersionObject = json.app_versions[0]
+
+    let version
+
+    // need this check because hockeyapp handles build numbers differntly for iOS and Android
+    if (
+      !latestVersionObject.version.includes(latestVersionObject.shortversion)
+    ) {
+      version = `${latestVersionObject.shortversion}.${
+        latestVersionObject.version
+      }`
+    } else {
+      version = latestVersionObject.version
+    }
+
+    return this.constructor.render({ version: version })
+  }
+
+  // Metadata
+  static get defaultBadgeData() {
+    return { label: 'hockeyapp' }
+  }
+
+  static get category() {
+    return 'version'
+  }
+
+  static get url() {
+    return {
+      base: 'hockeyapp/v',
+      format: '(.+)/(.+)',
+      capture: ['apptoken', 'appid'],
+    }
+  }
+
+  static get examples() {
+    return [
+      {
+        exampleUrl: 'hockeyapp/simple',
```

Comment:
I've set up a staging deploy for this at https://shields-staging-pr-2055.herokuapp.com/#/ and this example give me an invalid message: ![](https://shields-staging-pr-2055.herokuapp.com/hockeyapp/v/hockeyapp/simple.svg) https://shields-staging-pr-2055.herokuapp.com/hockeyapp/v/hockeyapp/simple.svg Can you take a look. The example should return a valid response

Reply:
Same issue here as with the test case that hits the live api - which account to use for this?